psycopg3/transaction.py

```diff
@@ -0,0 +1,86 @@
+"""
+Transaction context managers returned by Connection.transaction()
+"""
+
+# Copyright (C) 2020 The Psycopg Team
+
+# TODO: Should this module be _underscored since it's implementation detail?
+from types import TracebackType
+from typing import Optional, Type, TYPE_CHECKING
+
+from .pq import TransactionStatus
+
+if TYPE_CHECKING:
+    from .connection import Connection
+
+
+class Transaction:
+    def __init__(self, conn: "Connection") -> None:
+        self._conn = conn
+        self._outer_transaction = False
+        self._savepoint_name: Optional[str] = None
+        self._original_autocommit: bool
+
+    @property
+    def connection(self) -> "Connection":
+        return self._conn
+
+    def __enter__(self) -> None:
+        with self._conn.lock:
+            if self._conn.pgconn.transaction_status == TransactionStatus.IDLE:
+                self._outer_transaction = True
+                self._savepoint_name = None
+
+                self._original_autocommit = self._conn.autocommit
+                if self._conn._autocommit:
```

Comment:
So, because of what pointed out below, I don't think you need to change this, and I don't think that changing `_autocommit` is desirable. However, if it was, you don't need this if.